File: Python/Arrays & Hashing/Top-K-Frequent-Elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        frequencyMap = {}
        frequencyList = [[] for i in range(len(nums) + 1)]

        for num in nums:
            frequencyMap[num] = frequencyMap.get(num, 0) + 1

        for num, count in frequencyMap.items():
            frequencyList[count].append(num)

        result = []

        for i in range(len(frequencyList) - 1, 0, -1):
            for n in frequencyList[i]:
                result.append(n)
                if len(result) == k:
                    return result

File: Go/Arrays & Hashing/Top-K-Frequent-Elements.go
package main

func main() {

}

func topKFrequent(nums []int, k int) []int {
	frequencyMap := make(map[int]int, len(nums))
	frequencySlice := make([][]int, len(nums)+1)

	for _, num := range nums {
		frequencyMap[num]++
	}

	for num, count := range frequencyMap {
		frequencySlice[count] = append(frequencySlice[count], num)
	}

	
	result := []int{}

	for i := len(frequencySlice) - 1; i > 0; i-- {
		result = append(result, frequencySlice[i]...)
		if len(result) == k {
			break
		}
	}

	return result
}
